app/controllers/products_controller.rb

```diff
@@ -0,0 +1,85 @@
+class ProductsController < ApplicationController
+
+  before_action :set_product, only:[:show, :edit, :update, :destroy]
+
+  def index
+    @products = Category.find(params[:category_id]).products
+  end
+
+  # GET /products/1
+  # GET /products/1.json
+  def show
+    @category = Category.find(params[:category_id])
```

Comment:
We could have moved it to before filter but it's OK for now, we will refactor it later.